arch/rv32i/src/lib.rs

```diff
@@ -62,6 +63,11 @@ pub extern "C" fn _start() {
     use core::arch::asm;
     unsafe {
         asm! ("
+            // Disable linker relaxation for code that sets up GP so that this doesn't
+            // get turned into `mv gp, gp`.
+            .option push
+            .option norelax
+
```

Comment:
Can you move this below the comment "Set the global pointer..." since this part of that operation?